storage/src/state/ledger.rs

```diff
@@ -1629,6 +1639,13 @@ impl<N: Network> TransactionState<N> {
         Err(anyhow!("Commitment {} is missing in storage", commitment))
     }
 
+    /// Returns all the records ciphertexts in the ledger.
+    fn get_ciphertexts(&self) -> impl Iterator<Item = N::RecordCiphertext> + '_ {
+        self.transitions
+            .values()
+            .flat_map(|(_, _, transition)| transition.ciphertexts().cloned().collect::<Vec<N::RecordCiphertext>>())
```

Comment:
Can we remove the `.cloned()` and change the return type to `Iterator<Item = &N::RecordCiphertext>` (or a `Cow::Borrowed` if the compiler complains)?